design/baremetal-operator/provisioner-passthrough.md

```diff
@@ -0,0 +1,64 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Add provisioner passthrough API to BareMetalHost
+
+## Status
+
+implementable
+
+## Summary
+
+Add a new interface so that it is possible to configure any provisioner specific options.
+
+## Motivation
+
+In some circumstances it is desirable to use some provisioner specific features that
+can't be exposed to generic operator interface, e.g. ironic ansible deploy driver,
+ironic vendor passthrough options etc.
+
+### Goals
+
+Expand the Operator API to make it possible to expand provisioners without
+furhter API changes.
+
+### Non-Goals
+
+Do not try to describe all known provisioner specific options in the Operator API.
+
+## Proposal
+
+Add a option to the BareMetalHost spec which indicates additional options to
+provisioner:
+
+```yaml
+  spec:
+    privisionerPassthrough:
+      ironic:
+        driver: ansible
+        ansible_extra:
+          foo: bar
+```
+
+### Work Items
+
+- Add generic passthrough parameter to BareMetalHost.Spec.
+- Implement any provisioner specific options as needed.
```

Comment:
I'd rather have this proposal define its complete scope rather than relying on "as needed". Otherwise we cannot even determine if this is the best way to address the use cases.